src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs

```diff
@@ -174,6 +174,8 @@ internal Template ProcessResourcesAndOutputs(Template template)
                 {
                     AddResourceMapping(resourceInfo.expandedPath, resourceInfo.resource.Path);
                 }
+
+                resourceInfo.resource.Type.Value = resourceNameAndType.Split(" ")[1];
```

Comment:
This looks like it replaces the type of the child resource with the full type definition that would be required of a top-level resource.

If that's the case, why is there a bunch of string manipulation looking for the short form of a resource type in `JsonPathResolver`?  Shouldn't we expect to always be trying to resolve a fully-typed string, and be comparing it against fully-typed strings?

Reply:
The changes on that file were all related to the issue I didn't fix in the original version of the PR

"For a rule that specifies "resourceType": "Microsoft.Web/sites/siteextensions", resolve resource type should check for child resources in resources of type "Microsoft.Web/sites". Because the "Microsoft.Web/sites/siteextensions" child resource will have its type specified as simply "siteextensions", resource type selection needs to be intelligent enough to understand the full type of the template resources when searching for the type specified in the rule."

Even if the type siteextensions now would be Microsoft.Web/sites/siteextensions, we still need to look through Microsoft.Web/sites first